impl.go

```diff
@@ -305,6 +320,33 @@ func validReceiver(recv string) bool {
 	return err == nil
 }
 
+// commentsBefore returns true if comments precede the field. It returns false
+// otherwise.
+func commentsBefore(field *ast.Field, cg []*ast.CommentGroup) bool {
+	if len(cg) > 0 {
+		return cg[0].Pos() < field.Pos()
+	}
+	return false
+}
+
+// flattenCommentMap flattens the comment map to a string. This function must be
+// used at the point when m is expected to have a single element.
```

Comment:
Nit: If you would, please put the line break after the period. See https://rhodesmill.org/brandon/2012/one-sentence-per-line/ for context and reasoning.

The comment as written is ok. If you want to tighten it up, you could write just `m must have exactly one element.`.

Let's enforce this by starting the function with

```go
if len(m) != 1 {
  panic("flattenCommentMap expects comment map of length 1")
}
```

Reply:
done.